components/clp-py-utils/clp_py_utils/clp_config.py

```diff
@@ -300,17 +305,97 @@ class Queue(BaseModel):
     host: str = "localhost"
     port: int = 5672
 
-    username: typing.Optional[str]
-    password: typing.Optional[str]
+    username: Optional[str]
+    password: Optional[str]
 
 
-class ArchiveOutput(BaseModel):
+class S3Config(BaseModel):
+    region_name: str
+    bucket: str
+    key_prefix: str
+
+    access_key_id: Optional[str] = None
+    secret_access_key: Optional[str] = None
+
+    @validator("region_name")
+    def validate_region_name(cls, field):
+        if field == "":
+            raise ValueError("region_name is not provided")
+        return field
+
+    @validator("bucket")
+    def validate_bucket(cls, field):
+        if field == "":
+            raise ValueError("bucket is not provided")
+        return field
+
+    @validator("key_prefix")
+    def validate_key_prefix(cls, field):
+        if field == "":
+            raise ValueError("key_prefix is not provided")
+        if not field.endswith("/"):
+            raise ValueError(r'key_prefix must end with "/"')
+        return field
+
+
+class FSStorage(BaseModel):
+    type: Literal[StorageType.FS.value] = StorageType.FS.value
     directory: pathlib.Path = pathlib.Path("var") / "data" / "archives"
+
+    @validator("directory")
+    def validate_directory(cls, field):
+        if "" == field:
+            raise ValueError("directory can not be empty")
+        return field
+
+    def make_config_path_absolute(self, clp_home: pathlib.Path):
```

Comment:
This should be `make_config_paths_absolute` since it's meant to make _any_ config paths contained in this object absolute (in this case, there's only one, but that may not always be the case and it's not the case in other objects where we have this method).

Reply:
Sure,
but most other config has "make_config_path_absolute" as method name though